test_functools.py

```diff
@@ -1,25 +1,30 @@
-import os
-import itertools
-import time
+from __future__ import annotations
+
 import copy
-import random
 import functools
+import itertools
+import os
 import platform
+import random
+import time
+from typing import Literal, TypeVar
 from unittest import mock
 
 import pytest
+
 from jaraco.classes import properties
+from jaraco.functools import Throttler, method_cache, retry, retry_call
 
-from jaraco.functools import Throttler, method_cache, retry_call, retry
+_T = TypeVar("_T")
 
 
 class TestThrottler:
     @pytest.mark.xfail(
-        os.environ.get('GITHUB_ACTIONS')  # type: ignore
+        os.environ.get('GITHUB_ACTIONS', False)
```

Comment:
I don't understand this change. Not passing a default for `.get` will return `None` when the key is not present... and `None` operates perfectly well as a `bool` (`False`). I'd prefer to rely on the default `None` as the false value.

I can see the error is:

```
23: error: Argument 1 to "__call__" of "_XfailMarkDecorator" has incompatible type "str | bool | None"; expected "str | bool"  [arg-type]
```

Well, that's annoying. It seems that even though `bool(None) is False`, a literal bool is required as a parameter to `xfail`. That seems like a bug in xfail's type spec.

After all, why does it accept `str` as an input?

Oh. So I went and looked at [the docs](https://docs.pytest.org/en/stable/how-to/skipping.html#condition-parameter). Although they don't explicitly state why this function accepts both a bool and a str, the examples suggest that if a str is used, it should be an expression that evaluates to a bool.

So really, the type checker is catching a real bug - this value should be converted to a bool and not left as a string.

Fixed in ca6026b.

Reply:
Your change makes sense as long as you only care about the presence of env var (which I would assume so for `GITHUB_ACTIONS` in a test)